Gems/ROS2/Code/Source/RobotImporter/URDF/JointsMaker.cpp

```diff
@@ -31,7 +30,7 @@ namespace ROS2
         // {1,0,0} to a vector given by the URDF joint need to be found. Heavily suboptimal element in this conversion is needed of
         // converting the unit quaternion to Euler vector.
         const AZ::Vector3 o3de_joint_dir{ 1.0, 0.0, 0.0 };
-        const AZ::Vector3 joint_axis = URDF::TypeConversions::ConvertVector3(joint->axis);
+        const AZ::Vector3 joint_axis = URDF::TypeConversions::ConvertVector3(joint->Axis()->Xyz());
```

Comment:
In case the joint has a fixed type, the `Axis` component [is removed](https://github.com/gazebosim/sdformat/blob/7f63d022a48d3a224a3b7a777316c3822525c7b2/src/parser_urdf.cc#L3153), which causes segfault here (`joint->Axis()` returns `nullptr`).

Reply:
Yes, I noticed that in the [new UnitTest](https://github.com/o3de/o3de-extras/blob/682f8e894465c4462f6175db2a67e8fef3815221/Gems/ROS2/Code/Tests/UrdfParserTest.cpp#L308-L356) for verifying that the `URDFPreserveFixedJoints` option didn't merge links together also returned a nullptr for the `sdf::Joint::Axis()`, so the test was changed to only check the sdf::Joint::Type()` to see it if was FIXED.

I missed updating the JointMaker class with that change and address that issue.